Sources/Private/CoreAnimation/Animations/ShapeAnimation.swift

```diff
@@ -82,19 +76,48 @@ extension Trim {
   // MARK: Fileprivate
 
   /// The `strokeStart` and `strokeEnd` keyframes to apply to a `CAShapeLayer`
-  ///  - `CAShapeLayer` requires that `strokeStart` be less than `strokeEnd`.
-  ///  - Since this isn't a requirement in the Lottie schema, there are
-  ///    some animations that have `strokeStart` and `strokeEnd` flipped.
-  ///  - If we detect that this is the case for this specific `Trim`, then
-  ///    we swap the start/end keyframes to match what `CAShapeLayer` expects.
-  fileprivate func caShapeLayerKeyframes()
-    -> (strokeStart: KeyframeGroup<Vector1D>, strokeEnd: KeyframeGroup<Vector1D>)
-  {
+  fileprivate func caShapeLayerKeyframes(context: LayerAnimationContext) throws
+    -> (strokeStart: KeyframeGroup<Vector1D>, strokeEnd: KeyframeGroup<Vector1D>) {
+    let strokeStart: KeyframeGroup<Vector1D>
+    let strokeEnd: KeyframeGroup<Vector1D>
+
+    // CAShapeLayer requires strokeStart to be less than strokeEnd. This
+    // isn't required by the Lottie schema, so some animations may have
+    // strokeStart and strokeEnd flipped. If we detect this is the case,
+    // then swap them.
     if startValueIsAlwaysGreaterThanEndValue() {
-      return (strokeStart: end, strokeEnd: start)
+      strokeStart = end
+      strokeEnd = start
     } else {
-      return (strokeStart: start, strokeEnd: end)
+      strokeStart = start
+      strokeEnd = end
     }
+
+    // Adjust the keyframes to account for trim offsets if possible
+    let adjustedStrokeStart = try adjustKeyframesForTrimOffsets(
+      strokeKeyframes: strokeStart.keyframes,
+      offsetKeyframes: offset.keyframes,
+      context: context)
+
+    let adjustedStrokeEnd = try adjustKeyframesForTrimOffsets(
+      strokeKeyframes: strokeEnd.keyframes,
+      offsetKeyframes: offset.keyframes,
+      context: context)
+
+    // Validate the adjusted keyframes and fallback on original if invalid
+    if
+      (adjustedStrokeStart + adjustedStrokeEnd).contains(where: {
+        $0.value.cgFloatValue < 0 || $0.value.cgFloatValue > 100
+      }) {
+      try context.logCompatibilityIssue("""
+        The CoreAnimation rendering engine doesn't support Trim offsets
```

Comment:
Can we make this error more specific, so it describes the specific Trim configuration that's unsupported? For example:

```suggestion
        The Core Animation rendering engine doesn't support Trim offsets with values outside of the range [0, 1].
```